channeldb/payment_control.go

```diff
@@ -143,20 +146,24 @@ func (p *PaymentControl) InitPayment(paymentHash lntypes.Hash,
 	return updateErr
 }
 
-// RegisterAttempt atomically records the provided PaymentAttemptInfo to the
+// RegisterAttempt atomically records the provided HTLCAttemptInfo to the
 // DB.
 func (p *PaymentControl) RegisterAttempt(paymentHash lntypes.Hash,
-	attempt *PaymentAttemptInfo) error {
+	attempt *HTLCAttemptInfo) error {
 
 	// Serialize the information before opening the db transaction.
 	var a bytes.Buffer
-	if err := serializePaymentAttemptInfo(&a, attempt); err != nil {
+	err := serializeHTLCAttemptInfo(&a, attempt)
+	if err != nil {
 		return err
 	}
-	attemptBytes := a.Bytes()
+	htlcInfoBytes := a.Bytes()
+
+	htlcIDBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(htlcIDBytes, attempt.AttemptID)
 
 	var updateErr error
-	err := p.db.Batch(func(tx *bbolt.Tx) error {
+	err = p.db.Batch(func(tx *bbolt.Tx) error {
```

Comment:
in the future should be good to investigate whether `Batch` or `Update` is more performant for our use case

Reply:
I think we should always do `Update` for now, until there is a real reason to optimize.

Reply:
agreed

Reply:
Converted to `Update`